cmake/StaticBuild.cmake

```diff
@@ -259,38 +271,50 @@ if(APPLE AND CMAKE_CROSSCOMPILING)
         set(apple_${f}flags_arch "${apple_${f}flags_arch} -isysroot ${CMAKE_OSX_SYSROOT}")
       endforeach()
     endif()
-elseif(gmp_build_host STREQUAL "")
+elseif(gmp_build_host STREQUAL "" AND CMAKE_LIBRARY_ARCHITECTURE)
     set(gmp_build_host "--build=${CMAKE_LIBRARY_ARCHITECTURE}")
 endif()
 
-
-
 build_external(libtasn1
-    CONFIGURE_EXTRA --disable-doc
+    CONFIGURE_COMMAND ./configure ${gmp_build_host} --disable-shared --disable-doc --prefix=${DEPS_DESTDIR} --with-pic
```

Comment:
We should probably rename `gmp_build_host` since it's now being used for other packages as well.

Reply:
I've renamed it to `build_host` instead